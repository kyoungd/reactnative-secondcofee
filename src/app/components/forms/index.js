import FormHeading from './FormHeading'
import SignInForm from './SignInForm'
import SignUpForm from './SignUpForm'
import ResetPasswordForm from './ResetPasswordForm'
import PersonalInformationForm from './PersonalInformationForm'
import ProfessionalExperienceForm from './ProfessionalExperienceForm'

export {
  FormHeading,
  SignInForm,
  SignUpForm,
  ResetPasswordForm,
  PersonalInformationForm,
  ProfessionalExperienceForm
}
