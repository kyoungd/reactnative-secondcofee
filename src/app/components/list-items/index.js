import FilterOptionListItem from './FilterOptionListItem'
import JobListItem from './JobListItem'
import ProfileDetailListItem from './ProfileDetailListItem'
import SavedJobListItem from './SavedJobListItem'
import SettingsListItem from './SettingsListItem'

export {
  FilterOptionListItem,
  JobListItem,
  ProfileDetailListItem,
  SavedJobListItem,
  SettingsListItem
}
