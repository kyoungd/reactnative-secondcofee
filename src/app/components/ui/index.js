import CustomActivityIndicator from './CustomActivityIndicator'
import CustomStatusBar from './CustomStatusBar'
import FormField from './FormField'
import RegularButton from './RegularButton'
import ActionButton from './ActionButton'
import SafeAreaViewWrapper from './SafeAreaViewWrapper'
import Separator from './Separator'
import ListEmptyComponent from './ListEmptyComponent'
import ListLoadingComponent from './ListLoadingComponent'
import MultiSliderComponent from './MultiSliderComponent'
import ImageProgressComponent from './ImageProgressComponent'
import ImageOverlayComponent from './ImageOverlayComponent'

export {
  CustomActivityIndicator,
  CustomStatusBar,
  FormField,
  RegularButton,
  ActionButton,
  SafeAreaViewWrapper,
  Separator,
  ListEmptyComponent,
  ListLoadingComponent,
  MultiSliderComponent,
  ImageProgressComponent,
  ImageOverlayComponent
}
