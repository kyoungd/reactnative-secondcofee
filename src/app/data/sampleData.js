export const SAMPLE_JOB_LISTINGS = [
  {
    photoURL: 'https://bit.ly/2HqJLtg',
    title: 'Terminal Operator',
    companyName: 'Shell - New Haven, CT',
    experience: '2-3 years',
    postedAt: '25 February 2019'
  },
  {
    photoURL: 'https://bit.ly/2HtM9Q2',
    title: 'Editorial, Marketing, and Publicity Intern',
    companyName: 'Penguin Random House LLC',
    experience: '4-5 years',
    postedAt: '2 Mar 2019'
  },
  {
    photoURL: 'https://bit.ly/2F0hlnb',
    title: 'Data Entry Associate',
    companyName: 'ProAct, Inc. - Gouverneur, NY',
    experience: '1-2 years',
    postedAt: '22 February 2019'
  },
  {
    photoURL: 'https://bit.ly/2HtM9Q2',
    title: 'Entertainment- Concert Crew',
    companyName: 'Kings Dominion',
    experience: '0-1 years',
    postedAt: '9 Mar 2019'
  }
]

export const SAMPLE_FILTER_OPTIONS = [
  {
    title: 'Field/Category',
    options: ['Tech', 'Banking', 'Fashion', 'Construction']
  },
  {
    title: 'Benefits',
    options: ['Remote', '401K', 'Paid Vacation', 'Pet Friendly']
  }
]

export const SAMPLE_SEARCH_RESULTS = [
  {
    photoURL: 'https://imgur.com/2hjRD3d.png',
    title: '2019 Machine Technology Internship',
    companyName: 'Martinez, CA',
    experience: '1-2 years',
    postedAt: '25 February 2019',
    isSaved: true,
    description:
      'Ungerboeck Software is looking to hire a talented UI Developer Intern for our St. Louisteam! This is a paid internship position that requires about 20 hours per week. This internship can start immediately or summer of 2019.',
    filters: [
      {
        title: 'Field/Category',
        options: ['Tech', 'Fashion']
      },
      {
        title: 'Benefits',
        options: ['Remote', 'Paid Vacation', 'Pet Friendly']
      }
    ],
    salary: '$3200/month'
  },
  {
    photoURL: 'https://imgur.com/2hjRD3d.png',
    title: 'Customer Service Representative',
    companyName: 'White Plains, NY',
    experience: '0-1 years',
    postedAt: '2 Mar 2019',
    description:
      "We’re looking for a rockstar A-player whose task is to help grow 's engaged users. You should have an expert-level understanding of and a deep knowledge of . The ideal candidate has experience with internet products and technologies, possesses the ability to make data-driven decisions",
    filters: [
      {
        title: 'Field/Category',
        options: ['Tech', 'Fashion']
      },
      {
        title: 'Benefits',
        options: ['Remote', 'Paid Vacation', 'Pet Friendly']
      }
    ],
    salary: '$3200/month'
  },
  {
    photoURL: 'https://imgur.com/2hjRD3d.png',
    title: 'Data Entry Associate',
    companyName: 'ProAct, Inc. - Gouverneur, NY',
    experience: '2-3 years',
    postedAt: '22 February 2019',
    isSaved: true,
    description:
      'You are a person who thrives in a large group, can ideate as part of a small team, and enjoys operating independently. Experience sitting through meetings that should have been an email and writing on whiteboards and/or multi-colored post-it notes is required. ',
    filters: [
      {
        title: 'Field/Category',
        options: ['Tech', 'Fashion']
      },
      {
        title: 'Benefits',
        options: ['Remote', 'Paid Vacation', 'Pet Friendly']
      }
    ],
    salary: '$3200/month'
  },
  {
    photoURL: 'https://imgur.com/2hjRD3d.png',
    title: 'UI/UX Developer',
    companyName: 'Ungerboeck Software International',
    experience: '4-5 years',
    postedAt: '9 Mar 2019',
    isSaved: true,
    description:
      "Our investors include Capital, magneto Partners, and . By joining you can have a(n) large impact on the future while working in a and environment. Depending on your role, you'll help solve scintillating problems ",
    filters: [
      {
        title: 'Field/Category',
        options: ['Tech', 'Fashion']
      },
      {
        title: 'Benefits',
        options: ['Remote', 'Paid Vacation', 'Pet Friendly']
      }
    ],
    salary: '$3200/month'
  }
]

export const SAMPLE_JOB_DETAIL = {
  photoURL: 'https://imgur.com/2hjRD3d.png',
  title: 'UI Developer Intern',
  companyName: "Ungerboeck Software International - O'Fallon, MO",
  description:
    "Our investors include Capital, magneto Partners, and . By joining you can have a(n) large impact on the future while working in a and environment. Depending on your role, you'll help solve scintillating problems ",
  filters: [
    {
      title: 'Field/Category',
      options: ['Tech', 'Fashion']
    },
    {
      title: 'Benefits',
      options: ['Remote', 'Paid Vacation', 'Pet Friendly']
    }
  ],
  salary: '$3200/month'
}
