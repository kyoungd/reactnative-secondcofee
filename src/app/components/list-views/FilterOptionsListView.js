import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components/native'
import { FilterOptionListItem } from '../list-items'

const Wrapper = styled.View`
  padding: 16px 16px 0px;
`

const ListWrapper = styled.View`
  flex-direction: row;
  flex-wrap: wrap;
`

export default function FilterOptionsListView({
  items,
  activeItems,
  onItemPress
}) {
  return (
    <Wrapper>
      <ListWrapper>
        {items.map((item, index) => (
          <FilterOptionListItem
            key={index}
            label={item}
            active={activeItems.includes(item)}
            onPress={() => onItemPress(item)}
          />
        ))}
      </ListWrapper>
    </Wrapper>
  )
}

FilterOptionsListView.propTypes = {
  items: PropTypes.array.isRequired,
  activeItems: PropTypes.array.isRequired,
  onItemPress: PropTypes.func.isRequired
}
