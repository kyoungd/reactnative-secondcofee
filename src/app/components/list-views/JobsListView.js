import React from 'react'
import PropTypes from 'prop-types'
import { FlatList, RefreshControl } from 'react-native'
import styled from 'styled-components/native'
import { ListLoadingComponent, ListEmptyComponent } from '../ui'
import { JobListItem } from '../list-items'

const Wrapper = styled.View`
  flex: 1;
`

export default class JobsListView extends React.Component {
  state = {
    refreshing: false
  }

  onListRefresh() {
    this.setState({ refreshing: true })

    setTimeout(() => {
      this.setState({ refreshing: false })
    }, 2000)
  }

  render() {
    const { loading, items, onItemPress } = this.props
    const { refreshing } = this.state

    return (
      <Wrapper>
        {loading && <ListLoadingComponent text="Loading jobs" />}

        {!loading && items.length === 0 && (
          <ListEmptyComponent text="No job listings available at the moment." />
        )}

        {!loading && items.length > 0 && (
          <FlatList
            style={{ flex: 1 }}
            data={items}
            keyExtractor={item => items.indexOf(item).toString()}
            renderItem={({ item }) => (
              <JobListItem {...item} onPress={() => onItemPress(item)} />
            )}
            refreshControl={
              <RefreshControl
                refreshing={refreshing}
                onRefresh={() => this.onListRefresh()}
              />
            }
          />
        )}
      </Wrapper>
    )
  }
}

JobsListView.propTypes = {
  loading: PropTypes.bool.isRequired,
  items: PropTypes.array.isRequired,
  onItemPress: PropTypes.func.isRequired
}
