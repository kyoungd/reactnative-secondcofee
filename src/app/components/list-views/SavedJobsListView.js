import React from 'react'
import PropTypes from 'prop-types'
import { FlatList, RefreshControl } from 'react-native'
import styled from 'styled-components/native'
import { ListLoadingComponent, ListEmptyComponent, Separator } from '../ui'
import { SavedJobListItem } from '../list-items'

const Wrapper = styled.View`
  flex: 1;
`

export default class SavedJobsListView extends React.Component {
  state = {
    refreshing: false
  }

  onListRefresh() {
    this.setState({ refreshing: true })

    setTimeout(() => {
      this.setState({ refreshing: false })
    }, 2000)
  }

  render() {
    const { loading, items, onItemPress, onItemRemovePress } = this.props
    const { refreshing } = this.state

    return (
      <Wrapper>
        {loading && <ListLoadingComponent text="Loading saved jobs" />}

        {!loading && items.length === 0 && (
          <ListEmptyComponent text="You have not saved any jobs yet." />
        )}

        {!loading && items.length > 0 && (
          <FlatList
            style={{ flex: 1 }}
            data={items}
            keyExtractor={item => items.indexOf(item).toString()}
            ItemSeparatorComponent={() => <Separator />}
            renderItem={({ item }) => (
              <SavedJobListItem
                {...item}
                onPress={() => onItemPress(item)}
                onRemovePress={() =>
                  items.length > 1 && onItemRemovePress(item)
                }
              />
            )}
            refreshControl={
              <RefreshControl
                refreshing={refreshing}
                onRefresh={() => this.onListRefresh()}
              />
            }
          />
        )}
      </Wrapper>
    )
  }
}

SavedJobsListView.propTypes = {
  loading: PropTypes.bool.isRequired,
  items: PropTypes.array.isRequired,
  onItemPress: PropTypes.func.isRequired,
  onItemRemovePress: PropTypes.func.isRequired
}
