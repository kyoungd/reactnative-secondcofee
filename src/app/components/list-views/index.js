import FilterOptionsListView from './FilterOptionsListView'
import JobsListView from './JobsListView'
import SavedJobsListView from './SavedJobsListView'

export { FilterOptionsListView, JobsListView, SavedJobsListView }
