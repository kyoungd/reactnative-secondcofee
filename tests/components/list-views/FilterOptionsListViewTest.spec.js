import React from 'react'
import { shallow } from 'react-native-testing-library'
import { FilterOptionsListView } from 'components/list-views'
import { SAMPLE_FILTER_OPTIONS } from 'data/sampleData'

test('Component renders correctly', () => {
  const { output } = shallow(
    <FilterOptionsListView
      items={SAMPLE_FILTER_OPTIONS[0].options}
      activeItems={['Tech', 'Remote']}
      onItemPress={item => item}
    />
  )
  expect(output).toMatchSnapshot()
})
