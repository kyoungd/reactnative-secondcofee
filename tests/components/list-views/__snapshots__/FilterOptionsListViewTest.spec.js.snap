// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders correctly 1`] = `
<ForwardRef>
  <ForwardRef>
    <FilterOptionListItem
      active={true}
      label="Tech"
      onPress={[Function]}
    />
    <FilterOptionListItem
      active={false}
      label="Banking"
      onPress={[Function]}
    />
    <FilterOptionListItem
      active={false}
      label="Fashion"
      onPress={[Function]}
    />
    <FilterOptionListItem
      active={false}
      label="Construction"
      onPress={[Function]}
    />
  </ForwardRef>
</ForwardRef>
`;
